app/components/errors/special-error-handler.tsx

```diff
@@ -0,0 +1,57 @@
+import { tw } from "~/utils/tw";
+import { SpecialErrorAdditionalData } from "./utils";
+import { useMemo } from "react";
+import { ErrorIcon } from ".";
+import { Button } from "../shared/button";
+
+export type SpecialErrorHandlerProps = {
+  className?: string;
+  style?: React.CSSProperties;
+  additionalData: SpecialErrorAdditionalData;
+};
+
+export default function SpecialErrorHandler({
+  className,
+  style,
+  additionalData,
+}: SpecialErrorHandlerProps) {
+  const content = useMemo(() => {
+    switch (additionalData.type) {
+      case "asset-from-other-org": {
+        return (
+          <div className="w-full md:max-w-screen-sm">
+            <h2 className="mb-2">Asset belongs to other workspace.</h2>
+            <p className="mb-4">
+              The asset you are trying to view belongs to a different workspace
+              you are part of. Would you like to switch to workspace{" "}
+              <span className="font-bold">
+                "{additionalData.assetOrganization.organization.name}"
+              </span>{" "}
+              to view the asset?
+            </p>
+
+            <Button>Switch workspace</Button>
```

Comment:
We can make this POST to `user.change-current-organization.ts` and update the action to be able to receive a `redirectTo` and redirect to that entry.  So we can directly give the url to redirect to after changing.